main.go

```diff
@@ -74,5 +82,14 @@ func main() {
 	chkfatal(err)
 	srv := makeHandler(&config, NewDaemon(state))
 	http.Handle("/", srv)
-	chkfatal(http.ListenAndServe(config.ListenAddr, nil))
+	if config.WebProtocol == "http" {
```

Comment:
Nitpick: I think this would be a bit more readble as a switch statement:

```go
switch config.WebProtocol {
case "http":
    ...
case "https":
    ...
default:
    ...
}